test/sharness/t0180-private-network.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/sh
+#
+# Copyright (c) 2015 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test private network feature"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+export LIBP2P_FORCE_PNET=1
+
+test_expect_success "daemon won't start with force pnet env but with no key" '
+	test_must_fail go-timeout 5 ipfs daemon > stdout 2>&1
+'
+
+unset LIBP2P_FORCE_PNET
+
+test_expect_success "daemon output incudes info about the reason" '
+	grep "private network was not configured but is enforced by the environment" stdout ||
+	test_fsh cat stdout
+'
+
+pnet_key() {
+	echo '/key/swarm/psk/1.0.0/'
+	echo '/bin/'
+	random 16
+}
+
+pnet_key > $IPFS_PATH/swarm.key
+
+LIBP2P_FORCE_PNET=1 test_launch_ipfs_daemon
+
+test_expect_success "set up iptb testbed" '
+	iptb init -n 5 -p 0 -f --bootstrap=none  &&
+	iptb for-each ipfs config --json Addresses.Swarm  '"'"'["/ip4/127.0.0.1/tcp/0"]'"'"'
+'
+
+set_key() {
+	node="$1"
+	keyfile="$2"
+
+	cp "$keyfile" "$IPTB_ROOT/$node/swarm.key"
+}
+
+pnet_key > key1
+pnet_key > key2
+
+set_key 1 key1
+set_key 2 key1
+
+set_key 3 key2
+set_key 4 key2
+
+unset LIBP2P_FORCE_PNET
+
+test_expect_success "start nodes" '
+	iptb start [0-4]
+'
+
+test_expect_success "try connecting node in public network with priv networks" '
+	iptb connect [1-4] 0
+'
+
+test_expect_success "node 0 (public network) swarm is empty" '
+	ipfsi 0 swarm peers &&
+	[ $(ipfsi 0 swarm peers | wc -l) -eq 0 ]
+'
+
+test_expect_success "try connecting nodes in different private networks" '
+	iptb connect 2 3
+'
+
+test_expect_success "node 3 (pnet 2) swarm is empty" '
+	ipfsi 3 swarm peers &&
+	[ $(ipfsi 3 swarm peers | wc -l) -eq 0 ]
```

Comment:
The iptb connect command waits for the connect call to finish, by then it is either for sure connected or not.